OSDOCS-14858:Update the z-stream RNs for 4.16.42 (#94442)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

//4.16.42
[id="ocp-4-16-42_{context}"]
=== RHSA-2025:8556 - {product-title} {product-version}.42 bug fix and security update

Issued: 11 June 2025

{product-title} release {product-version}.42 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2025:8556[RHSA-2025:8556] advisory. 

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.16.42 --pullspecs
----

[id="ocp-4-16-42-bug-fixes_{context}"]
==== Bug fixes

> * Previously, the console alerted users that compute nodes must be updated within 60 days when performing a control plane only update. With this update, the console no longer displays this invalid alert. (link:https://issues.redhat.com/browse/OCPBUGS-56858[OCPBUGS-56858])

Reviewer: ```suggestion
* Previously, the web console alerted users that compute nodes must be updated within 60 days when performing a control plane only update. With this update, the web console no longer displays this invalid alert. (link:https://issues.redhat.com/browse/OCPBUGS-56858[OCPBUGS-56858])
```
It is actually important to distinguish this console from other things, such as the hybrid cloud console.

* Previously, OpenShift Container Platform 4.15 and later versions managed by OpenShift Lifecycle Manager (OLM) were required to have the `olm.managed: "true"` label. In some cases, the solution failed to start and entered a `CrashLoopBackOff` state if the label was missing. The logs for this scenario were displayed as informative, which made it more challenging to identify the root cause. For this release, the log level is changed to error to make the issue clearer and easier to diagnose when the label is missing. (link:https://issues.redhat.com/browse/OCPBUGS-56358[OCPBUGS-56358])

* Previously, if the default proxy environment variables were set to null on build containers, some applications in the container would not run. With this release, the proxy environment variables are added to the build container only if they are defined and the default values are not null. (link:https://issues.redhat.com/browse/OCPBUGS-56354[OCPBUGS-56354])

> * Previously, disabling feature migration prevented computer network operations (CNO) from initiating software defined networking (SDN) live migration. With this release, CNO can trigger SDN live migration when feature migration is disabled. (link:https://issues.redhat.com/browse/OCPBUGS-56195[OCPBUGS-56195])

Author: You're correct. I mistakenly entered "computer network operations".

> * Previously, there was no event logged when an error occurred from failed conversion from ingress to route conversion. With this update, this error appear in the event logs. (link:https://issues.redhat.com/browse/OCPBUGS-56152[OCPBUGS-56152])

Reviewer: ```suggestion
* Previously, there was no event logged when an error occurred after a failed conversion from ingress to route. With this update, this error appear in the event logs. (link:https://issues.redhat.com/browse/OCPBUGS-56152[OCPBUGS-56152])
```

> * Previously, {azure-short} spot machines that were evicted before their node became ready could get stuck in the 'provisioned' state. With this release, {azure-short} spot instances now use a delete eviction policy. This policy ensures that the machines correctly move to the 'failed' state upon preemption. (link:https://issues.redhat.com/browse/OCPBUGS-56092[OCPBUGS-56092])

Reviewer: ```suggestion
* Previously, {azure-short} spot machines that were evicted before their node became ready could get stuck in the `provisioned` state. With this release, {azure-short} spot instances now use a delete eviction policy. This policy ensures that the machines correctly move to the `failed` state upon preemption. (link:https://issues.redhat.com/browse/OCPBUGS-56092[OCPBUGS-56092])
```

* Previously, Zscaler traffic scanning caused a pull-progress timeout in the CRI-O container engine, creating an infinite loop and image pull failure. As a result, users were unable to pull images. With this release, the pull progress timeout is now 20 seconds, disabling the timeout and progress output for image pulls and reducing the customer image pull timeout issues with Zscaler. (link:https://issues.redhat.com/browse/OCPBUGS-54665[OCPBUGS-54665])

* Previously, when the MAC-binding flows on {azure-short} expired, Open vSwitch (OVS) executed a controller action to reinstall that flow. If the OVN controller was slow right after the flows were deleted, the new flows were not installed in time. This delay reduced the dataplane traffic. With this release, dataplane traffic is not reduced because the new flows are correctly installed by the controller action. (link:https://issues.redhat.com/browse/OCPBUGS-53151[OCPBUGS-53151])

>  * Previously, during the restart of OVN-Kubernetes containers, routes for the internal `ovn-k8-mp0` interface were removed and readded, resulting in temporary traffic outage. With this release, the traffic paths flowing across the `ovn-k8s-mp0` interface are not interrupted and the routes are not removed during ovn-kubernetes pod restart. (link:https://issues.redhat.com/browse/OCPBUGS-52503[OCPBUGS-52503])

Reviewer: ```suggestion
 * Previously, during the restart of OVN-Kubernetes containers, routes for the internal `ovn-k8-mp0` interface were removed and readded, resulting in temporary traffic outage. With this release, the traffic paths flowing across the `ovn-k8s-mp0` interface are not interrupted and the routes are not removed during the `ovn-kubernetes` pod restart. (link:https://issues.redhat.com/browse/OCPBUGS-52503[OCPBUGS-52503])
```
